pgmpy/models/FunctionalBayesianNetwork.py

```diff
@@ -0,0 +1,188 @@
+import networkx as nx
+import numpy as np
+import pandas as pd
+import pyro
+
+from pgmpy.factors.hybrid import FunctionalCPD
+from pgmpy.global_vars import logger
+from pgmpy.models import BayesianNetwork
+
+
+class FunctionalBayesianNetwork(BayesianNetwork):
+    """
+    A Functional Gaussian Bayesian Network is a Bayesian Network,
+    whose variables can be discrete or continuous, and where all of the CPDs
+    are defined by FunctionalCPD.
+
+    An important result is that the Functional Bayesian Networks
+    provide flexible representation for the class of multiples uni/multi-variate
+    distributions.
+    """
+
+    def add_cpds(self, *cpds):
+        """
+        Add Functional CPD (Conditional Probability Distribution)
+        to the Bayesian Network.
+
+        Parameters
+        ----------
+        cpds  :  instances of FunctionalCPD
+            List of FunctionalCPDs which will be associated with the model
+
+        Examples
+        --------
+        >>> from pgmpy.factors.hybrid import FunctionalCPD
+        >>> from pgmpy.models import FunctionalBayesianNetwork
+        >>> import numpy as np
+
+        >>> model = FunctionalBayesianNetwork([("x1", "x2"), ("x2", "x3")])
+        >>> cpd1 = FunctionalCPD("x1", lambda _: np.random.normal(0, 1))
+        >>> cpd2 = FunctionalCPD("x2", lambda parent: np.random.normal(parent["x1"] + 2.0, 1), parents=["x1"])
+        >>> cpd3 = FunctionalCPD("x3", lambda parent: np.random.normal(parent["x2"] + 0.3, 2), parents=["x2"])
+        >>> model.add_cpds(cpd1, cpd2, cpd3)
+
+        """
+        for cpd in cpds:
+            if not isinstance(cpd, FunctionalCPD):
+                raise ValueError("Only FunctionalCPD can be added.")
+
+            if set(cpd.variables) - set(cpd.variables).intersection(set(self.nodes())):
+                raise ValueError("CPD defined on variable not in the model", cpd)
+
+            for prev_cpd_index in range(len(self.cpds)):
+                if self.cpds[prev_cpd_index].variable == cpd.variable:
+                    logger.warning(f"Replacing existing CPD for {cpd.variable}")
+                    self.cpds[prev_cpd_index] = cpd
+                    break
```

Comment:
Should this be `continue`? `break` would exit out of the `for cpd in cpds` loop.